### ISSUE TITLE
unexpected end of file     at Zlib._handle.onerror (zlib.js:363:17)

### ISSUE THREAD

**Issue by GabLeRoux** [tldr-pages/tldr#319]
```
npm install -g tldr
/usr/local/bin/tldr -> /usr/local/lib/node_modules/tldr/bin/tldr
/usr/local/lib
└─┬ tldr@1.0.1
  ├── async@0.9.2
  ├── colors@1.0.3
  ├── commander@2.3.0
  ├─┬ cpr@0.3.3
  │ └── graceful-fs@3.0.8
  ├── deep-extend@0.2.11
  ├── lodash@2.4.2
  ├── marked@0.3.5
  ├─┬ mkdirp@0.5.1
  │ └── minimist@0.0.8
  ├── ms@0.6.2
  ├─┬ request@2.44.0
  │ ├── aws-sign2@0.5.0
  │ ├── bl@0.9.4
  │ ├── caseless@0.6.0
  │ ├── forever-agent@0.5.2
  │ ├─┬ form-data@0.1.4
  │ │ ├─┬ combined-stream@0.0.7
  │ │ │ └── delayed-stream@0.0.5
  │ │ └── mime@1.2.11
  │ ├─┬ hawk@1.1.1
  │ │ ├── boom@0.4.2
  │ │ ├── cryptiles@0.2.2
  │ │ ├── hoek@0.9.1
  │ │ └── sntp@0.2.4
  │ ├─┬ http-signature@0.10.1
  │ │ ├── asn1@0.1.11
  │ │ ├── assert-plus@0.1.5
  │ │ └── ctype@0.5.3
  │ ├── json-stringify-safe@5.0.1
  │ ├── mime-types@1.0.2
  │ ├── node-uuid@1.4.3
  │ ├── oauth-sign@0.4.0
  │ ├── qs@1.2.2
  │ ├── stringstream@0.0.5
  │ ├── tough-cookie@2.2.0
  │ └── tunnel-agent@0.4.1
  ├── rimraf@2.2.8
  ├─┬ unhtml@0.1.0
  │ ├── html5-entities@0.5.1
  │ └── quotemeta@0.0.0
  ├─┬ unzip2@0.2.5
  │ ├─┬ binary@0.3.0
  │ │ ├── buffers@0.1.1
  │ │ └─┬ chainsaw@0.1.0
  │ │   └── traverse@0.3.9
  │ ├─┬ fstream@0.1.31
  │ │ └── inherits@2.0.1
  │ ├── match-stream@0.0.2
  │ ├─┬ pullstream@0.4.1
  │ │ ├── over@0.0.5
  │ │ └── slice-stream@1.0.0
  │ ├─┬ readable-stream@1.0.33
  │ │ ├── core-util-is@1.0.1
  │ │ ├── isarray@0.0.1
  │ │ └── string_decoder@0.10.31
  │ └── setimmediate@1.0.4
  └── wrench@1.5.8
```
Then I run update command.
```
tldr --update
```
I get
```
Updating...
Error: unexpected end of file
    at Zlib._handle.onerror (zlib.js:363:17)
```

```
tldr --version
1.0.1
```

**Comment by R3s1stanc3** [comment]
I've got the same problem. Any solutions jet?

**Comment by rprieto** [comment]
Does it happen all the time or is it intermittent?

**Comment by igorshubovych** [comment]
It happens for me from time to time too

**Comment by R3s1stanc3** [comment]
For me it happenes every time, I try to update tldr

**Comment by izolate** [comment]
Getting a similar error:

    Updating...
    Error: invalid distance too far back
        at Zlib._handle.onerror (zlib.js:363:17)

**Comment by EthanOrange** [comment]
ops...
it happenes for me  every time.

**Comment by GabLeRoux** [comment]
It's not hapening to me anymore, Can't tell what changes exactly, but in fact, it stays stuck on the command, I don't know if this is a normal behaviour (I don't think so);

```
tldr update
Updating...
Done

```

**Comment by igorshubovych** [comment]
It could be because of Github API issues then. Because NodeJS client was not updated.

**Comment by GabLeRoux** [comment]
Now getting this,  unsure if it's related.
```
tldr update
Updating...
Error: invalid signature: 0xb68c7950
    at /usr/local/lib/node_modules/tldr/node_modules/unzip2/lib/parse.js:63:13
    at processImmediate [as _immediateCallback] (timers.js:383:17)
```

**Comment by igorshubovych** [comment]
@GabLeRoux,

Please update tldr-node-client to 1.1.0:

`npm update -g`

And then run pages update again

`tldr --update`

**Comment by GabLeRoux** [comment]
> tldr --update
> Updating...
> Done

Client closed by itself with no problem, seems to work like a charm, great work
